### ISSUE TITLE
Can't upload files with `-` in the filename.

### ISSUE THREAD

**Issue by mrDarcyMurphy** [scitran/core#184]
Stack trace:

```
uwsgi_1          | 2016-03-07 17:06:42      scitran.api                  base.py    46:DEBU looked up cached token in 5ms
uwsgi_1          | 2016-03-07 17:06:42      scitran.api                  base.py   145:DEBU from None darcymurphy@invenshure.com POST /api/projects/56dd9be37fec7f32c1a96037/files {}
uwsgi_1          | 2016-03-07 17:06:42      scitran.api           liststorage.py    41:DEBU query {'_id': ObjectId('56dd9be37fec7f32c1a96037')}
uwsgi_1          | 2016-03-07 17:06:42      scitran.api            validators.py   152:DEBU {u'key_fields': [u'name'], u'required': [u'name', u'created', u'modified', u'size', u'hash'], u'additionalProperties': False, u'$schema': u'http://json-schema.org/draft-04/schema#', u'type': u'object', u'properties': {u'mimetype': {u'type': u'string'}, u'hash': {u'type': u'string'}, u'name': {u'type': u'string'}, u'created': {}, u'measurements': {u'uniqueItems': True, u'items': {u'type': u'string'}, u'type': u'array'}, u'modified': {}, u'instrument': {u'type': u'string'}, u'metadata': {u'type': u'object'}, u'type': {u'type': u'string'}, u'tags': {u'uniqueItems': True, u'items': {u'type': u'string'}, u'type': u'array'}, u'size': {u'type': u'integer'}}}
uwsgi_1          | 2016-03-07 17:06:42      scitran.api                  base.py   128:ERRO Traceback (most recent call last):
uwsgi_1          |   File "/usr/local/lib/python2.7/dist-packages/webapp2.py", line 570, in dispatch
uwsgi_1          |     return method(*args, **kwargs)
uwsgi_1          |   File "./api/handlers/listhandler.py", line 371, in post
uwsgi_1          |     return upload.process_upload(self.request, upload.Strategy.targeted, cont_name, _id)
uwsgi_1          |   File "./api/upload.py", line 103, in process_upload
uwsgi_1          |     field.size	 = os.path.getsize(field.path)
uwsgi_1          |   File "/usr/lib/python2.7/genericpath.py", line 49, in getsize
uwsgi_1          |     return os.stat(filename).st_size
uwsgi_1          | OSError: [Errno 2] No such file or directory: '/var/scitran/data/.tmpbox4uX/cisforcookiethatsgoodenoughforme.gif'
uwsgi_1          |
uwsgi_1          | ./api/tempdir.py:53: Warning: Implicitly cleaning up <TemporaryDirectory u'/var/scitran/data/.tmpbox4uX'>
uwsgi_1          |   Warning)
uwsgi_1          | [pid: 22|app: 0|req: 10/15] 192.168.99.1 () {56 vars in 1147 bytes} [Mon Mar  7 17:06:42 2016] POST /api/projects/56dd9be37fec7f32c1a96037/files => generated 139 bytes in 130 msecs (HTTP/1.1 500) 3 headers in 131 bytes (2 switches on core 0)
```

This doesn't happen for filenames that have spaces or underscores in them, only dashes. Looks like the dashes are being stripped from the filename (reference?).

`c-is-for-cookie-thats-good-enough-for-me.gif` becomes `cisforcookiethatsgoodenoughforme.gif` when it's looking up the file in the tmp directory.

**Comment by kofalt** [comment]
Haha, amazing. I think the problem is a sanitized filename that doesn't take effect everywhere.

**Comment by kofalt** [comment]
Solved in af55f6f
